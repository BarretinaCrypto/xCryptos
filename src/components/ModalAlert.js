import { bool, func, shape, string } from 'prop-types';
import React, { Component } from 'react';
import { Text, View } from 'react-native';
import { connect } from 'react-redux';
import { FormLabel, FormInput } from 'react-native-elements';
import { addAlertAction, removeAlertAction } from '../actions';
import { SHAPE, STYLE, THEME } from '../config';
import { ServiceAlerts } from '../services';
import Button from './Button';
import ButtonIcon from './ButtonIcon';
import Modal from './Modal';
import styles from './ModalAlert.style';

const { ALERT, CURRENCY } = SHAPE;

class ModalAlert extends Component {
  static navigationOptions({ navigation: { state } }) {
    const { currency = {}, _showAlert } = state.params || {};

    return {
      title: `${currency.name} Alerts`,
      headerRight: <ButtonIcon icon="add" onPress={() => { _showAlert(); }} />,
    };
  }

  constructor(props) {
    super(props);
    this.state = {
      item: undefined,
      refreshing: false,
    };
    this._onChange = this._onChange.bind(this);
    this._onSubmit = this._onSubmit.bind(this);
  }

  componentWillReceiveProps() {
    this.setState({ item: undefined });
  }

  _onChange(field, value) {
    const { item = {} } = this.state;

    this.setState({
      item: { ...item, [field]: parseFloat(value) },
    });
  }

  async _onSubmit() {
    const {
      addAlert, alert, removeAlert, currency: { symbol }, onClose, token,
    } = this.props;
    const { item = {} } = this.state;

    this.setState({ refreshing: true });
    if (alert) await ServiceAlerts.remove(alert).then(removeAlert);
    else await ServiceAlerts.add({ ...item, currency: symbol, token }).then(addAlert);
    this.setState({ refreshing: false });
    onClose();
  }

  render() {
    const { _onChange, _onSubmit } = this;
    const {
      alert, currency: { symbol, usd }, onClose, visible,
    } = this.props;
    const { item = alert, refreshing } = this.state;
    const { low, high } = item || {};

    const inputProps = {
      containerStyle: styles.fieldReset,
      keyboardType: 'numeric',
      underlineColorAndroid: THEME.PRIMARY,
    };

    return (
      <Modal
        title={alert ? 'Alert' : 'New Alert'}
        onClose={onClose}
        visible={visible}
      >
        <View style={[STYLE.CENTERED, STYLE.ROW]}>
          <Text style={styles.symbol}>$</Text>
          <Text style={styles.price}>{usd}</Text>
          <View style={[STYLE.CHIP, styles.chip]}>
            <Text style={styles.chipCaption}>{symbol}</Text>
          </View>
        </View>
        <View style={[STYLE.ROW]}>
          <View style={styles.fieldset}>
            <FormLabel labelStyle={[styles.fieldReset]}>Low</FormLabel>
            <FormInput
              autoFocus
              defaultValue={item && low && low.toString()}
              editable={!alert}
              inputStyle={styles.input}
              onChangeText={_onChange.bind(null, 'low')} //eslint-disable-line
              {...inputProps}
            />
          </View>
          <View style={styles.fieldset}>
            <FormLabel labelStyle={[styles.fieldReset, styles.labelRight]}>High</FormLabel>
            <FormInput
              defaultValue={item && high && high.toString()}
              editable={!alert}
              inputStyle={[styles.input, styles.inputRight]}
              onChangeText={_onChange.bind(null, 'high')} //eslint-disable-line
              {...inputProps}
            />
          </View>
        </View>
        <Button
          caption={item && item.currency ? 'Delete' : 'Create'}
          disabled={(!alert && (!low || !high || low > usd || high < usd)) || refreshing}
          onPress={() => { _onSubmit(); }}
          style={[STYLE.MODAL_BUTTON, styles.modalButton]}
        />
      </Modal>
    );
  }
}

ModalAlert.propTypes = {
  addAlert: func,
  alert: shape(ALERT),
  currency: shape(CURRENCY),
  visible: bool,
  onClose: func,
  removeAlert: func,
  token: string,
};

ModalAlert.defaultProps = {
  addAlert() {},
  alert: undefined,
  currency: undefined,
  visible: false,
  onClose() {},
  removeAlert() {},
  token: undefined,
};

const mapStateToProps = state => ({
  token: state.token,
});

const mapDispatchToProps = dispatch => ({
  addAlert: alert => alert && dispatch(addAlertAction(alert)),
  removeAlert: alert => alert && dispatch(removeAlertAction(alert)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ModalAlert);
