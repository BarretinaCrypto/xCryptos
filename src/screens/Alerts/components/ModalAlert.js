import { bool, func, shape, string } from 'prop-types';
import React, { Component } from 'react';
import { Text, View } from 'react-native';
import { connect } from 'react-redux';
import { addAlertAction, removeAlertAction } from '../../../actions';
import { Amount, Button, Input, Modal } from '../../../components';
import { SHAPE, STYLE } from '../../../config';
import { ServiceAlerts } from '../../../services';
import styles from './ModalAlert.style';

const { ALERT, COIN, SETTINGS } = SHAPE;

const Fieldset = ({
  label, disabled, onChange, price, right, value = 0, // eslint-disable-line
}) => (
  <View style={[styles.fieldset, right && styles.alignRight]}>
    <View style={STYLE.ROW}>
      <Text style={styles.label}>{label}</Text>
      {
        value > 0 &&
        <Text style={[styles.label, styles.percent]}>
          {` (${parseInt(((value * 100) / price) - 100, 10)}%)`}
        </Text>
      }
    </View>
    {
      disabled
      ?
        <Amount style={styles.input} value={value} />
      :
        <Input
          autoFocus={label === 'low'}
          defaultValue={value ? value.toString() : undefined}
          onChangeText={newValue => onChange(label, newValue)}
          style={[styles.input, right && styles.inputRight]}
        />
    }
  </View>
);

class ModalAlert extends Component {
  constructor(props) {
    super(props);
    this.state = {
      item: undefined,
      refreshing: false,
    };
    this._onChange = this._onChange.bind(this);
    this._onSubmit = this._onSubmit.bind(this);
  }

  componentWillReceiveProps() {
    this.setState({ item: undefined });
  }

  _onChange(field, value) {
    const { item = {} } = this.state;
    this.setState({
      item: { ...item, [field]: parseFloat(value) },
    });
  }

  async _onSubmit() {
    const {
      props: {
        addAlert, alert, removeAlert, onClose, token,
        coin: { coin },
        settings: { currency },
      },
      state: {
        item: { low, high } = {},
      },
    } = this;

    this.setState({ refreshing: true });
    if (alert) {
      await ServiceAlerts.remove(alert).then(removeAlert);
    } else {
      await ServiceAlerts.add({
        coin, currency, low, high, token,
      }).then(addAlert);
    }
    this.setState({ refreshing: false });
    onClose();
  }

  render() {
    const {
      _onChange, _onSubmit,
      props: {
        alert, coin: { price }, onClose, visible,
      },
      state: {
        item: { low, high } = alert || {}, refreshing,
      },
    } = this;
    const invalid = !alert && (!low || !high || low > price || high < price);

    return (
      <Modal title={!alert ? 'New Alert' : 'Alert'} onClose={onClose} visible={visible}>
        <View style={[STYLE.CENTERED, STYLE.LIST_ITEM, styles.content]}>
          <Amount style={styles.price} value={price} />
        </View>
        <View style={[STYLE.ROW, STYLE.LIST_ITEM, styles.content]}>
          <Fieldset disabled={alert} label="low" onChange={_onChange} value={low} price={price} />
          <Fieldset disabled={alert} label="high" onChange={_onChange} right value={high} price={price} />
        </View>
        <View style={STYLE.MODAL_FOOTER}>
          <Button
            caption={alert ? 'Delete' : 'Create'}
            disabled={invalid || refreshing}
            onPress={_onSubmit}
            style={styles.button}
          />
        </View>
      </Modal>
    );
  }
}

ModalAlert.propTypes = {
  addAlert: func,
  alert: shape(ALERT),
  coin: shape(COIN).isRequired,
  onClose: func,
  removeAlert: func,
  settings: shape(SETTINGS),
  token: string,
  visible: bool,
};

ModalAlert.defaultProps = {
  addAlert() {},
  alert: undefined,
  onClose() {},
  removeAlert() {},
  settings: {},
  token: undefined,
  visible: false,
};

const mapStateToProps = ({ settings, token }) => ({
  settings,
  token,
});

const mapDispatchToProps = dispatch => ({
  addAlert: alert => alert && dispatch(addAlertAction(alert)),
  removeAlert: alert => alert && dispatch(removeAlertAction(alert)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ModalAlert);
