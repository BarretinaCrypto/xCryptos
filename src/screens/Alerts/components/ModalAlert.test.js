import 'react-native';
import React from 'react';
import { ModalAlert } from './ModalAlert';
import renderer from 'react-native-test-utils';

jest.mock('../../../components/Amount', () => 'Amount');
jest.mock('../../../services/alerts', () => ({
  add: () => Promise.resolve(),
  remove: () => Promise.resolve(),
}));

const COIN = { price: 5 };
const ALERT = { low: 1, high: 10 };

describe('<ModalAlert />', () => {
  describe('when alert is new', () => {
    it('renders by default', () => {
      const tree = renderer(<ModalAlert coin={COIN} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders the correct title', () => {
      const component = renderer(<ModalAlert coin={COIN} />);
      const modal = component.query('Modal');
      
      expect(modal.props.title).toEqual('New Alert');
    });

    it('renders the correct caption in <Button />', () => {
      const component = renderer(<ModalAlert coin={COIN} />);
      const button = component.query('Button');
      
      expect(button.props.caption).toEqual('Create');
    });

    describe('when low is empty', () => {
      it('disables submit button', () => {
        const component = renderer(<ModalAlert coin={COIN} />);
        component.instance.setState({
          item: { low: null, high: 5 }
        });

        const button = component.query('Button');
        expect(button.props.disabled).toBe(true);
      });
    });

    describe('when low is bigger than price', () => {
      it('disables submit button', () => {
        const component = renderer(<ModalAlert coin={COIN} />);
        component.instance.setState({
          item: { low: 10, high: 5 }
        });

        const button = component.query('Button');
        expect(button.props.disabled).toBe(true);
      });
    });

    describe('when high is empty', () => {
      it('disables submit button', () => {
        const component = renderer(<ModalAlert coin={COIN} />);
        component.instance.setState({
          item: { low: 2, high: null }
        });

        const button = component.query('Button');
        expect(button.props.disabled).toBe(true);
      });
    });

    describe('when hight is smaller than price', () => {
      it('disables submit button', () => {
        const component = renderer(<ModalAlert coin={COIN} />);
        component.instance.setState({
          item: { low: 2, high: 3 }
        });

        const button = component.query('Button');
        expect(button.props.disabled).toBe(true);
      });
    });
  });

  describe('when alert already exists', () => {
    it('renders by default', () => {
      const tree = renderer(<ModalAlert coin={COIN} alert={ALERT} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders the correct title', () => {
      const component = renderer(<ModalAlert coin={COIN} alert={ALERT} />);
      const modal = component.query('Modal');
      
      expect(modal.props.title).toEqual('Alert');
    });

    it('renders the correct caption in <Button />', () => {
      const component = renderer(<ModalAlert coin={COIN} alert={ALERT} />);
      const button = component.query('Button');
      
      expect(button.props.caption).toEqual('Delete');
    });

    it('disables fieldsets', () => {
      const component = renderer(<ModalAlert coin={COIN} alert={ALERT} />);
      const fieldsets = component.queryAll('Fieldset');

      fieldsets.forEach(fieldset => expect(fieldset.props.disabled).toBeTruthy());
    });
  });

  describe('when refreshing is true', () => {
    it('disables submit button', () => {
      const component = renderer(<ModalAlert coin={COIN} />);
      component.instance.setState({
        item: { low: 2, high: 16 },
        refreshing: true
      });

      const button = component.query('Button');
      expect(button.props.disabled).toBe(true);
    });
  });

  describe('._onChange', () => {
    it('sets the state with the correct value', () => {
      const component = renderer(<ModalAlert coin={COIN} />).instance;

      expect(component.state.item).toBeUndefined();

      component._onChange('low', 66);

      expect(component.state.item.low).toEqual(66);
    });
  });

  describe('._onSubmit', () => {
    it('sets refreshing to true', () => {
      const component = renderer(<ModalAlert coin={COIN} />).instance;
      component._onSubmit();
      expect(component.state.refreshing).toBe(true);
    });

    describe('when there is an alert', () => {
      it('calls removeAlert prop', async () => {
        const removeSpy = jest.fn();
        const component = renderer(<ModalAlert coin={COIN} alert={ALERT} removeAlert={removeSpy} />).instance;
        
        await component._onSubmit();
        
        expect(removeSpy).toHaveBeenCalled();
      });
    });

    describe('when alert is new', () => {
      it('calls addAlert prop', async () => {
        const addSpy = jest.fn();
        const component = renderer(<ModalAlert coin={COIN} addAlert={addSpy} />).instance;
        
        await component._onSubmit();
        
        expect(addSpy).toHaveBeenCalled();
      });
    });
  });
});
