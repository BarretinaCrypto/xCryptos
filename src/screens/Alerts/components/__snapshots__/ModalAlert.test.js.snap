// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalAlert /> when alert already exists renders by default 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "height": "100%",
        "justifyContent": "flex-end",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View
      keyboardVerticalOffset={0}
      onLayout={[Function]}
      style={undefined}
    >
      <View
        style={
          Array [
            undefined,
            Object {
              "bottom": 0,
            },
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "opacity": 1,
              "paddingVertical": 16,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                },
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "marginBottom": 16,
                  "paddingHorizontal": 16,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 1,
                  "fontSize": 20,
                  "fontWeight": "700",
                  "textAlign": "center",
                }
              }
            >
              Alert
            </Text>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Image
                source={1}
                style={
                  Array [
                    Object {
                      "height": 28,
                      "marginLeft": 10,
                      "marginRight": 10,
                      "resizeMode": "cover",
                      "width": 28,
                    },
                    Object {
                      "marginLeft": -28,
                      "marginRight": 0,
                      "opacity": 0.5,
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "borderTopColor": "rgba(0, 0, 0, 0.1)",
                "borderTopWidth": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderBottomColor": "rgba(0, 0, 0, 0.1)",
                    "borderBottomWidth": 1,
                    "paddingHorizontal": 16,
                    "paddingVertical": 10,
                  },
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 10,
                  },
                ]
              }
            >
              <Amount
                style={
                  Object {
                    "fontSize": 48,
                    "fontWeight": "200",
                  }
                }
                value={5}
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderBottomColor": "rgba(0, 0, 0, 0.1)",
                    "borderBottomWidth": 1,
                    "paddingHorizontal": 16,
                    "paddingVertical": 10,
                  },
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 10,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "width": "50%",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.5)",
                        "fontSize": 16,
                      }
                    }
                  >
                    low
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.5)",
                          "fontSize": 16,
                        },
                        Object {
                          "fontSize": 12,
                        },
                      ]
                    }
                  >
                     (-80%)
                  </Text>
                </View>
                <Amount
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 24,
                    }
                  }
                  value={1}
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "width": "50%",
                    },
                    Object {
                      "alignItems": "flex-end",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.5)",
                        "fontSize": 16,
                      }
                    }
                  >
                    high
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.5)",
                          "fontSize": 16,
                        },
                        Object {
                          "fontSize": 12,
                        },
                      ]
                    }
                  >
                     (100%)
                  </Text>
                </View>
                <Amount
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 24,
                    }
                  }
                  value={10}
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "padding": 16,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#ed1e79",
                        "borderRadius": 32,
                        "display": "flex",
                        "paddingHorizontal": 16,
                        "paddingVertical": 10,
                      },
                      Object {
                        "paddingVertical": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#ffffff",
                          "fontSize": 16,
                          "fontWeight": "700",
                        },
                        Array [],
                      ]
                    }
                  >
                    Delete
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`<ModalAlert /> when alert is new renders by default 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "height": "100%",
        "justifyContent": "flex-end",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View
      keyboardVerticalOffset={0}
      onLayout={[Function]}
      style={undefined}
    >
      <View
        style={
          Array [
            undefined,
            Object {
              "bottom": 0,
            },
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "opacity": 1,
              "paddingVertical": 16,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                },
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "marginBottom": 16,
                  "paddingHorizontal": 16,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 1,
                  "fontSize": 20,
                  "fontWeight": "700",
                  "textAlign": "center",
                }
              }
            >
              New Alert
            </Text>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Image
                source={1}
                style={
                  Array [
                    Object {
                      "height": 28,
                      "marginLeft": 10,
                      "marginRight": 10,
                      "resizeMode": "cover",
                      "width": 28,
                    },
                    Object {
                      "marginLeft": -28,
                      "marginRight": 0,
                      "opacity": 0.5,
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "borderTopColor": "rgba(0, 0, 0, 0.1)",
                "borderTopWidth": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderBottomColor": "rgba(0, 0, 0, 0.1)",
                    "borderBottomWidth": 1,
                    "paddingHorizontal": 16,
                    "paddingVertical": 10,
                  },
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 10,
                  },
                ]
              }
            >
              <Amount
                style={
                  Object {
                    "fontSize": 48,
                    "fontWeight": "200",
                  }
                }
                value={5}
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderBottomColor": "rgba(0, 0, 0, 0.1)",
                    "borderBottomWidth": 1,
                    "paddingHorizontal": 16,
                    "paddingVertical": 10,
                  },
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 10,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "width": "50%",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.5)",
                        "fontSize": 16,
                      }
                    }
                  >
                    low
                  </Text>
                </View>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  autoFocus={true}
                  defaultValue={undefined}
                  keyboardType="numeric"
                  onChangeText={[Function]}
                  placeholderTextColor={undefined}
                  style={
                    Array [
                      Object {
                        "fontSize": 20,
                        "paddingBottom": 10,
                        "paddingLeft": 2.5,
                        "paddingRight": 2.5,
                        "paddingTop": 10,
                        "width": "100%",
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 0.5)",
                      },
                      Array [
                        Object {
                          "color": "black",
                          "fontSize": 24,
                        },
                        undefined,
                      ],
                    ]
                  }
                  underlineColorAndroid="transparent"
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "width": "50%",
                    },
                    Object {
                      "alignItems": "flex-end",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.5)",
                        "fontSize": 16,
                      }
                    }
                  >
                    high
                  </Text>
                </View>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  autoFocus={false}
                  defaultValue={undefined}
                  keyboardType="numeric"
                  onChangeText={[Function]}
                  placeholderTextColor={undefined}
                  style={
                    Array [
                      Object {
                        "fontSize": 20,
                        "paddingBottom": 10,
                        "paddingLeft": 2.5,
                        "paddingRight": 2.5,
                        "paddingTop": 10,
                        "width": "100%",
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 0.5)",
                      },
                      Array [
                        Object {
                          "color": "black",
                          "fontSize": 24,
                        },
                        Object {
                          "textAlign": "right",
                        },
                      ],
                    ]
                  }
                  underlineColorAndroid="transparent"
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "padding": 16,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#ed1e79",
                        "borderRadius": 32,
                        "display": "flex",
                        "paddingHorizontal": 16,
                        "paddingVertical": 10,
                      },
                      Object {
                        "paddingVertical": 16,
                      },
                      Object {
                        "backgroundColor": "black",
                        "opacity": 0.35,
                      },
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#ffffff",
                          "fontSize": 16,
                          "fontWeight": "700",
                        },
                        Array [],
                      ]
                    }
                  >
                    Create
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
